### PR TITLE
ci: install rust with pinned version

### PR DESCRIPTION
**Pull request by alvicsam**
PR installs rust in CI with pinned version

close https://github.com/polkadot-fellows/runtimes/issues/240

- [x] Does not require a CHANGELOG entry

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -177,19 +175,16 @@ jobs:
...
 
       - name: Install nightly toolchain
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly
-          target: wasm32-unknown-unknown
+        run: |
+          rustup toolchain install "nightly-${{ env.RUST_NIGHTLY_VERSION }}" --profile minimal --component rust-src
+          rustup target add wasm32-unknown-unknown --toolchain "nightly-${{ env.RUST_NIGHTLY_VERSION }}"
```
Comment (ggwpez) [inline .github/workflows/test.yml:187]: Not sure if we even need nightly for this job as it only uses `cargo test`?

**Review by ggwpez — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (alvicsam) [reply .github/workflows/test.yml]: It seems we don't, removed in https://github.com/polkadot-fellows/runtimes/pull/247/commits/9373fde97864b53aff55bee83d01d7e47e7160c3

File: .github/workflows/fmt.yml [hunk 2]
```diff
@@ -6,21 +6,19 @@ on:
   pull_request:
   workflow_dispatch:
 
+env:
+  RUST_NIGHTLY_VERSION: 2024-03-14 # 1.78.0-nightly
```
Comment (bkchr) [inline .github/workflows/fmt.yml:10]: Could we please put these env variables into one file? So we don't always need to touch all the files. We can then also relax the review requirements on this file to the same level as the normal source files.

**Review by bkchr — APPROVED** [review]

File: .github/workflows/fmt.yml (hunk 2 above)
Comment (bkchr) [reply .github/workflows/fmt.yml]: Ty!

**Review by andresilva — APPROVED** [review]

**Pull request merged**